kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -149,7 +149,7 @@ namespace Kratos
 			// by using the ComputePlaneApproximation utility. Otherwise, the distance is computed using
 			// the plane defined by the 3 intersection points.
 			auto &r_geometry = rElement1.GetGeometry();
-			const bool do_plane_approx = (n_cut_edges == r_geometry.WorkingSpaceDimension()) ? false : true;
+			const bool do_plane_approx = (n_cut_edges == r_geometry.WorkingSpaceDimension() || (int_pts_vector.size() < TDim)) ? false : true;
```

Comment:
Here I check if the number of intersection points is at least equal to Dim.

In test 7, the tetrahedron with one edge laying on the horizontal plane at z = 0.0, had n_cut_edges > NDIM but only 2 intersection points, so the approximation utility was giving a plane which was not the horizontal one